### DISCUSSION TITLE
6.13-rc1 can boot

### DISCUSSION THREAD

**Discussion by hakotani-o** [Joshua-Riek/ubuntu-rockchip#1135] [Ideas]
6.13-rc1 can boot.
But audio not work.
![2024-11-28 22-46-18](https://github.com/user-attachments/assets/8bf47b38-b857-493b-aa1a-200333c8cb87)
![Screenshot From 2024-11-29 08-41-03](https://github.com/user-attachments/assets/02c3c033-7af6-43d5-b50f-6acc95fed999)

**Comment by jimmyhon** [comment]
Are you trying to use the es8388 (analog audio) on the Orange Pi 5?

If so, you'll want to open up `alsamixer` and toggle on the Headphone widget.

**Reply by hakotani-o** [reply]
Thank you for your interest.
Yes,i want to use es8388(analog audio).
I try alsamixer , there was no sound.

Thank you.

**Comment by stevepringle** [comment]
How did you go about building this? Some instructions would be great if you wouldn’t mind sharing your knowledge. :-)

**Reply by hakotani-o** [reply]
Thank you for your interest.
Attached is a file summarizing the steps.
[build.txt](https://github.com/user-attachments/files/18091391/build.txt)

Thank you.

**Reply by stevepringle** [reply]
Many thanks for that. I’ll give it a go this weekend if I get chance :-)

**Reply by nobara256** [reply]
@hakotani-o Thank you so much for giving a simple build procedure, I was able to successfully build the upstream kernel for the first time.

**Reply by hakotani-o** [reply]
Thank you for trying out my build script.
Please modify the contents of () for the kernel version.

\# set version
cat << EOF > debian.rockchip/changelog
linux-rockchip (6.13.0-rc3.20241221) oracular; urgency=medium

  *Initial release.

 -- Your name <your@mail.address> Wed, 11 Dec 2024 20:48:48 +0900
EOF

**Reply by ShuoYangRobotics** [reply]
Hi @hakotani-o Thanks for creating this! I am new to Linux kernel building. I used your build.txt to successfully build kernel 6.13. Now I should let my Orange Pi to load the new kernel?

**Comment by stevepringle** [comment]
Thank you for your script, I have managed to build the kernel (6.13.0-rc4). How do I get hold of the build scripts to build an Ubuntu release for oracular (the ones I have form Joshuas site dont include oracular)

**Reply by hakotani-o** [reply]
How to create a disk image.

    1.Install build tools. ( It may contain things you don't need. )
    sudo apt-get install -y build-essential gcc-aarch64-linux-gnu bison \
    qemu-user-static qemu-system-arm qemu-efi-aarch64 u-boot-tools binfmt-support \
    debootstrap flex libssl-dev bc rsync kmod cpio xz-utils fakeroot parted \
    udev dosfstools uuid-runtime git-lfs device-tree-compiler python3 \
    python-is-python3 fdisk bc debhelper python3-pyelftools python3-setuptools \
    python3-pkg-resources swig libfdt-dev libpython3-dev \
    git fakeroot build-essential ncurses-dev xz-utils libssl-dev bc flex libelf-dev bison gawk

    2.ubuntu-rockchip build tool download.
    git clone --depth=1 https://github.com/Joshua-Riek/ubuntu-rockchip.git
    
    3.run build.sh
      cd ubuntu-rockchip
      mkdir build
      cp kernel-image-files build
   \# Please change --board=Your board. 
         sudo ./build.sh --board=orangepi-5-plus --suite=oracular --flavor=desktop

    4.If the build is successful
    The disk image files are present in the "images" directory.

Thank you.

**Reply by stevepringle** [reply]
Many thanks, that’s just what I needed:-)

**Comment by hakotani-o** [comment]
Hello.
I have created a build script for ubuntu mainline.

 1. run.sh setup
Two scripts are run.
The parameters are
working directory name board type dtb filename.

2. The disk image is created in the images directory.

3. The kernel version is tested with torvalds v6.13-rc3.
You can specify the kernel version by changing the parameter git clone -b in u-bootstrap.sh.
The kernel configuration file is overlay/config.common.ubuntu.
Note that the cpu temperature will be high during kernel build.

4. Operation was confirmed by writing the disk image to SD-CARD and booting.
It takes less than 1 minute until the screen appears.
It also supports grub boot with edk2-rk3588, but the grub menu did not appear unless the ESC key is pressed.

5. The snap application is not installed.
Please install it after boot.

The contents of this script are mostly based on
copied from Joshua Riek's build system. 
Thanks to Joshua Riek.

[ubuntu-mainline.tar.gz](https://github.com/user-attachments/files/18257359/ubuntu-mainline.tar.gz)
Thank you.

**Comment by nobara256** [comment]
Have you checked the GPU status on 6.13.0-rc4? I can see the `CONFIG_DRM_PANTHOR=m` and `CONFIG_DRM_GPUVM=m` properly configured under `debian.rockchip/config/config.common.ubuntu`. But my server images don't seem to have `panthor` working as intended. In `vulkaninfo` I can still see it as `llvmpipe`. Maybe I am missing something in the server images or something requires to be added in the device-tree?

**Reply by hakotani-o** [reply]
Sorry for the late reply.
Thank you for providing such great information.
If you install the graphic driver from below, it will be recognized as Mali-G610 Panfrost.
https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers

As for stability, I have no information.
The kernel is also under development, so if you're feeling adventurous, give it a try.

Thank you.

**Reply by nobara256** [reply]
Thanks, I can finally see some relevant output.

```
Instance Layers: count = 2
--------------------------
VK_LAYER_MESA_device_select Linux device selection layer 1.4.303  version 1
VK_LAYER_MESA_overlay       Mesa Overlay layer           1.4.303  version 1

Devices:
========
GPU0:
	apiVersion         = 1.0.303
	driverVersion      = 25.0.0
	vendorID           = 0x13b5
	deviceID           = 0xa8670000
	deviceType         = PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU
	deviceName         = Mali-G610 (Panfrost)
	driverID           = DRIVER_ID_MESA_PANVK
	driverName         = panvk
	driverInfo         = Mesa 25.0~git2412310600.e112e2~oibaf~o (git-e112e2b 2024-12-31 oracular-oibaf-ppa)
	conformanceVersion = 0.0.0.0
GPU1:
	apiVersion         = 1.3.303
	driverVersion      = 0.0.1
	vendorID           = 0x10005
	deviceID           = 0x0000
	deviceType         = PHYSICAL_DEVICE_TYPE_CPU
	deviceName         = llvmpipe (LLVM 19.1.1, 128 bits)
	driverID           = DRIVER_ID_MESA_LLVMPIPE
	driverName         = llvmpipe
	driverInfo         = Mesa 25.0~git2412310600.e112e2~oibaf~o (git-e112e2b 2024-12-31 oracular-oibaf-ppa) (LLVM 19.1.1)
	conformanceVersion = 1.3.1.1
	deviceUUID         = 6d657361-3235-2e30-7e67-697432343100
	driverUUID         = 6c6c766d-7069-7065-5555-494400000000
```

`GPU0` does appear as `Mali-G610 (Panfrost)` and even that as a `PHYSICAL_DEVICE_TYPE_INTEGRATED_GPU`. Although the apiVersion seems to be ancient, I think it's a great start.

Thank you so much.

**Comment by hakotani-o** [comment]
I updated ubuntu-mainline.tar.xz because an error occurred with add-apt-repository.

[ubuntu-mainline.tar.gz](https://github.com/user-attachments/files/18308690/ubuntu-mainline.tar.gz)

**Reply by Eternalightwithin** [reply]
I'm about to try this for the first time. Pardon the noob question. In your build.txt file, you have
mkdir oracular
cd oracular
git clone --depth 5 -b oracular https://github.com/Joshua-Riek/linux-rockchip.git

Then further down....
mkdir 6.13
cd 6.13
git clone https://github.com/torvalds/linux.git

    Is one the mainline 6.13 and the other one (JRiek) patches?
    What is the current set ver number? I can't find the vtag on Tolvald's page.
    Should I stick with 6.13.0-rc2.20241209 or try 6.14.x or current 6.15-RC4?
    What's Ubuntu-mainline.tar.gz for? Does it replace build.txtor do I use it in OPi as a config?  Going to bed. will think about it more after work.


P.S. I went ahead and tried it.
I'm using 25.04 Plucky on AMD
I pasted your build.txt in terminal and was good until

Debug: binary-headers
dh_installchangelogs -plinux-rockchip-headers-6.13.0-rc2
dh_installchangelogs: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
dh_installdocs -plinux-rockchip-headers-6.13.0-rc2
dh_installdocs: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
dh_compress -plinux-rockchip-headers-6.13.0-rc2
dh_compress: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
dh_fixperms -plinux-rockchip-headers-6.13.0-rc2
dh_fixperms: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
dh_installdeb -plinux-rockchip-headers-6.13.0-rc2
dh_installdeb: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
flock -w 60 /home/eternalight7/6.13/linux/debian/.LOCK dh_gencontrol -plinux-rockchip-headers-6.13.0-rc2
dh_gencontrol: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
dh_md5sums -plinux-rockchip-headers-6.13.0-rc2
dh_md5sums: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
dh_builddeb -plinux-rockchip-headers-6.13.0-rc2
dh_builddeb: warning: All requested packages have been excluded (e.g. via a Build-Profile or due to architecture restrictions).
cc -o debian/scripts/fix-filenames debian/scripts/fix-filenames.c
Debug: /home/eternalight7/6.13/linux/debian/stamps/stamp-prepare-tree-rockchip
install -d /home/eternalight7/6.13/linux/debian/build/build-rockchip
touch /home/eternalight7/6.13/linux/debian/build/build-rockchip/ubuntu-build
[ "false" != 'true' ] && true || \
	rsync -a --exclude debian --exclude debian.master --exclude debian.rockchip * /home/eternalight7/6.13/linux/debian/build/build-rockchip
if [ -e /home/eternalight7/6.13/linux/debian.rockchip/config/config.common.ubuntu ]; then \
	cat /home/eternalight7/6.13/linux/debian.rockchip/config/config.common.ubuntu /home/eternalight7/6.13/linux/debian.rockchip/config/amd64/config.common.amd64 /home/eternalight7/6.13/linux/debian.rockchip/config/amd64/config.flavour.rockchip > /home/eternalight7/6.13/linux/debian/build/build-rockchip/.config; \
else \
	python3 debian/scripts/misc/annotations --export --arch amd64 --flavour rockchip | sed -e 's/.*CONFIG_VERSION_SIGNATURE.*/CONFIG_VERSION_SIGNATURE="Ubuntu 6.13.0-rc2.20241209-rockchip make[1]: Entering directory '/home/eternalight7/6.13/linux' 6.15.0-rc5 make[1]: Leaving directory '/home/eternalight7/6.13/linux'"/' > /home/eternalight7/6.13/linux/debian/build/build-rockchip/.config; \
fi
cat: /home/eternalight7/6.13/linux/debian.rockchip/config/amd64/config.common.amd64: No such file or directory
cat: /home/eternalight7/6.13/linux/debian.rockchip/config/amd64/config.flavour.rockchip: No such file or directory
make: *** [debian/rules.d/2-binary-arch.mk:28: /home/eternalight7/6.13/linux/debian/stamps/stamp-prepare-tree-rockchip] Error 1
eternalight7@Xperimental:~/6.13/linux$ 

Thank you,

Eternalight

**Reply by hakotani-o** [reply]
Hi eternalightwithin.

Sorry i not use this script.
Now i use my own customized build system.
You can find the scripts in my repository.
I omit features from the kernel config that I don't use.
For example wifi, BT, NFS, etc.

Thank you.

**Reply by hakotani-o** [reply]
Please add 4 export , before build kernel.

export $(dpkg-architecture -aarm64)
export CROSS_COMPILE=aarch64-linux-gnu-
export CC=aarch64-linux-gnu-gcc
export LANG=C

Thank you.

**Comment by cscd98** [comment]
Has anyone tried 6.15 that was just released?

**Reply by stevepringle** [reply]
![Screenshot From 2025-05-27 10-20-43](https://github.com/user-attachments/assets/7e1952e6-6ed8-4186-b2bf-de54b80863ca)
![Screenshot From 2025-05-27 10-21-47](https://github.com/user-attachments/assets/994cd1c2-0f19-493d-a2d9-25450fbc12cb)
Yes,I’ve got it running on a radxa  rock 5 itx board with ubuntu 25.04. I had to modify the .dts file for the rock-5-itx board to enable the hdmi sound and use the ppa from https://launchpad.net/~oibaf/+archive/ubuntu/graphics-drivers to get the Vulkan drivers working. 

The only issue I found is that the ubuntu installation process is very slow, with error messages saying that the installation process is not responding, but clicking on the keep waiting button eventually results in the process completing OK.

The hdmi graphics resolutions supported are limited to a few options at the moment, but it seems very stable with hdmi graphics and sound working OK. Vkcube works fine and shows it is using the on board Mali 610 integrated graphics chip. I used a modified version of hakotani-o’s build scripts to do the build. It’s also running a Plex server and streaming shows such as Farscape without any judder at a screen resolution of 2560x1440@60Hz and the hdmi sound is working fine. 4K works but only at a 30Hz refresh rate. According to the Collabora mainline kernel site support for 4k@60Hz is a work in progress as is the NPU support. (https://gitlab.collabora.com/hardware-enablement/rockchip-3588/notes-for-rockchip-3588/-/blob/main/mainline-status.md)

Build file attached if it is of interest to anyone:-
[build-ubuntu-mainline.tar.gz](https://github.com/user-attachments/files/20454405/build-ubuntu-mainline.tar.gz)

**Comment by hakotani-o** [comment]
My personal build tools for opi-5.

kernel: 6.15 limited features
ubuntu: 25.04 plucky
mesa: 25.1 local build

[my-orangepi-5-build.tar.gz](https://github.com/user-attachments/files/20532795/my-orangepi-5-build.tar.gz)

Thank you.

**Comment by Joshua-Riek** [comment]
I'll eventueventually get to updating everything, though my plan is to only release server images with mainline Linux. I have no time frame, though.

**Reply by gattohub** [reply]
I'm trying to build a rock5b image for Ubuntu 25.04 with mainline using the scripts, but I'm not able to. Can anyone help me?

**Reply by hakotani-o** [reply]
Please try building it in a noble-arm64 environment.

**Reply by gattohub** [reply]
It's the same if I upgrade from 24.04 server image?
The "kernel-image-files build" command fails.

**Reply by hakotani-o** [reply]
Please try this method.
The operating environment can be either noble or plucky, but please use arm64. Cross-compilation is not supported.

1. Download ubuntu-mainline2.tar.gz
2. tar zxvf ubuntu-mainline2.tar.gz
3. cd ubuntu-mainline2
4. ./run.sh

Thank you.

[ubuntu-mainline2.tar.gz](https://github.com/user-attachments/files/21327953/ubuntu-mainline2.tar.gz)

**Reply by ChiefPete** [reply]
hakotani-o,

Thanks for providing this build script.

I have a Indiedroid Nova that I would like to build a mainline image for. What files/variables would I need to modified in your build scripts to build an image for the Nova.

Thanks for any assistance.

**Reply by hakotani-o** [reply]
Please edit run.sh.

sudo  ./u-bootstrap.sh arm64 nova  indiedroid-nova-rk3588s 2>&1|tee log1
sudo  ./disk_image.sh arm64 nova  indiedroid-nova-rk3588s 2>&1|tee log2

Thank you.
